docs/collections.md

```diff
@@ -0,0 +1,271 @@
+# Collections
+
+Robo provides task collections as a means of making error detection and recovery easier. When Robo tasks are added to a collection, their execution is deferred until the `$collection->run()` method is called.  When using collections, a Robo script will go through three phases:
+
+1. Determine which tasks will need to be run, and create a collection.
+  - Assign values to variables.
+  - Do not alter the state of the system.
+2. Create the necessary tasks and add them to the collection.
+  - Pass variables calculated in the first phase to task parameters.
+3. Run the tasks via `$collection->run()`.
+  - Check and report errors once after `run()` returns.
+
+Following this pattern will keep your code linear and easy to understand.
+
+## Basic Collection Example
+
+A simple example is shown below:
+
+``` php
+<?php
+class RoboFile extends \Robo\Tasks
+{
+    function myOperation()
```

Comment:
I think we can have a use case and real command name for better understanding here